### DISCUSSION TITLE
Typing to loose on WhereUnique

### DISCUSSION THREAD

**Discussion by vdhpieter** [prisma/prisma#25902] [Q&A]
### Bug description

When using a findUnique on a model that has multiple unique fields we expected a TS error if that field was possibly undefined. In reality we didn't get an error.

### How to reproduce

See code below

### Prisma information

```prisma
model Address {
  id             String     @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  userId         String     @unique @map("user_id") @db.Uuid
  ...
}
```

```ts
await prisma.address.findUnique({
  where: { userId: user?.id },
}) // No TS error while it's clear for TS that userId could be undefined
```

### Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: v20.18.0

### Prisma Version

```
prisma                  : 5.22.0
@prisma/client          : 5.22.0
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.18.0
Query Engine (Node-API) : libquery-engine 605197351a3c8bdd595af2d2a9bc3025bca48ea2 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli 605197351a3c8bdd595af2d2a9bc3025bca48ea2 (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.22.0-44.605197351a3c8bdd595af2d2a9bc3025bca48ea2
Default Engines Hash    : 605197351a3c8bdd595af2d2a9bc3025bca48ea2
Studio                  : 0.503.0
```

**Comment by ludralph** [comment]
Hi @vdhpieter 

This was previously discussed in this [issue](https://github.com/prisma/prisma/issues/17723#issuecomment-1423106211). A quick summary is:
```
When  we encounter undefined, we strip it away from the query and treat it as if you never passed that information. 
```

But with the  introduction of the [`strictUndefinedChecks` Preview feature](https://www.prisma.io/docs/orm/prisma-client/special-fields-and-types/null-and-undefined) this behaviour is no longer supported. You should now get better type checking and runtime behavior for cases where fields might be undefined.

**Comment by vdhpieter** [comment]
@ludralph When I enable the Preview Feature I still don't get a TS error when writing the above, Should this be the case?

**Reply by ludralph** [reply]
Hi @vdhpieter 

`strictUndefinedChecks` primarily affect runtime behavior and not Typescript static type checking. Enabling it will throw an error at runtime for when user is undefined.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by vdhpieter** [comment]
@ludralph for me the discussion is not resolved, I still think the types could be stricter to catch these kind of issues before they are even ran.
